pkg/search/postgres/common.go

```diff
@@ -598,6 +604,30 @@ func standardizeSelectQueryAndPopulatePath(q *v1.Query, schema *walker.Schema, q
 	return parsedQuery, nil
 }
 
+func scopeContextToQuery(ctx context.Context, q *v1.Query) (*v1.Query, error) {
+	scopeQ, err := scoped.GetQueryForAllScopes(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if scopeQ == nil {
+		return q, nil
+	}
+	cloned := q.Clone()
```

Comment:
I feel like this section of code where we make a copy, save off bits, remove bits, add them back in, warrants a comment.  or maybe a `ScopedClone` function that hides some of this.